### PR TITLE
Added calc so cart notificaiton comes up correctly w/header margin

### PR DESCRIPTION
**Pull request by tyleralsbury**
**Update**

Refactored to put the margin on a different element, thus negating the need for a calc. Thanks @KaichenWang for the idea.

**Why are these changes introduced?**

Fixes a small bug where the cart notification would not emerge from the header properly if there was a margin set in the header settings.

To test, change the margin setting in the header and notice that the Cart Notification is always properly positioned.  Before, it would be X pixels too low, where X is the value of the setting.

Test both Desktop and Mobile because the setting is applied at a fraction on mobile.

**What approach did you take?**

Added a calc that accounts for the setting.

**Demo links**

- [Store](https://os2-demo.myshopify.com/?_ab=0&_fd=0&_sc=1&preview_theme_id=127107629078)
- [Editor](https://os2-demo.myshopify.com/admin/themes/127107629078/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: sections/header.liquid [hunk 1]
```diff
@@ -71,10 +71,18 @@
     margin-bottom: {{ section.settings.margin_bottom | times: 0.75 | round: 0 }}px;
   }
```
[the remark this answers is not in the document]
Comment (tyleralsbury) [reply sections/header.liquid:73]: That's interesting - it looks like what's happening is that you're ending up with the sticky header receding.  It looks like you're scrolling because your screen is smaller than the total size of the header + the notification.

If you test with a larger screen that shouldn't happen - I think a screen that small is an edge case.

**Comment by KaichenWang** [comment]
Could this be avoided in first place if the header's margin value is added to the section container instead?

In the schema, we could add a header-specific class
```json
"class": "section-header"
```

```css
{%- style -%}
  .section-header {
    margin-bottom: {{ section.settings.margin_bottom | times: 0.75 | round: 0 }}px;
  }

  @media screen and (min-width: 750px) {
    .section-header {
      margin-bottom: {{ section.settings.margin_bottom }}px;
    }
  }
{%- endstyle -%}
```

**Comment by tyleralsbury** [comment]
> Could this be avoided in the first place if the header's margin value is added to the section container instead?
> 
> In `header.liquid` schema, we could add a header-specific class
> 
> ```json
> "class": "section-header"
> ```
> 
> Then use that class to apply the bottom margin (instead of targeting `.header-wrapper`)
> 
> ```css
> {%- style -%}
>   .section-header {
>     margin-bottom: {{ section.settings.margin_bottom | times: 0.75 | round: 0 }}px;
>   }
> 
>   @media screen and (min-width: 750px) {
>     .section-header {
>       margin-bottom: {{ section.settings.margin_bottom }}px;
>     }
>   }
> {%- endstyle -%}
> ```

Good idea - I will give it a try and see.

File: sections/header.liquid (hunk 1 above)
[the remark this answers is not in the document]
Comment (sofiamatulis) [reply sections/header.liquid:73]: Yes! It was only when I had my console open

**Comment by tyleralsbury** [comment]
Love it, @KaichenWang - looks like it worked well!

**Review by sofiamatulis — APPROVED** [review]
Awesome 🎉

**Review by KaichenWang — APPROVED** [review]
